draft-irtf-cfrg-vdaf.md

```diff
@@ -1234,6 +1238,35 @@ terminal states are `Rejected`, which indicates that the report cannot be
 processed any further, and `Finished(out_share)`, which indicates that the
 Aggregator has recovered an output share `out_share`.
 
+For convenience, the methods described in this section are defined in terms of
+opaque byte strings. A compatible `Vdaf` MUST specify methods for encoding
+public shares, input shares, prep shares, and prep messages. Minimally:
+
+* `Vdaf.decode_public_share(agg_id: Unsigned, encoded: bytes) ->
+  Vdaf.PublicShare` decodes a public share, using the aggregator ID as
+  optional context.
+
+* `Vdaf.decode_input_share(agg_id: Unsigned, encoded: bytes) ->
+  Vdaf.InputShare` decodes an input share, using the aggregator ID as optional
+  context.
+
+* `Vdaf.encode_prep_share(prep_share: Vdaf.PrepShare) -> bytes` encodes a prep
+  share.
+
+* `Vdaf.decode_prep_share(prep_state: Vdaf.PrepState, encoded: bytes) ->
+  Vdaf.PrepShare` decodes a prep share, using the prep state as optional
+  context.
+
+* `Vdaf.encode_prep_msg(prep_msg: Vdaf.PrepMessage) -> bytes` encodes a prep
+  message.
+
+* `Vdaf.decode_prep_msg(prep_state: Vdaf.PrepState, encoded: bytes) ->
+  Vdaf.PrepMessage` decodes a prep message, using the prep state as optional
+  decoding context.
+
+Implementations of Prio3 and Poplar1 MUST use the encoding scheme specified in
+{{prio3-encode}} and {{poplar1-encode}} respectively.
```

Comment:
This MUST seems, unfortunately, necessary to me. One open question: how will the ping-pong API extend to other VDAFs that aren't in this document?